dev/snippets/lib/main.dart

```diff
@@ -33,44 +40,40 @@ void main(List<String> argList) {
     help: 'The type of snippet to produce.',
   );
   parser.addOption(
-    'template',
+    _kTemplateOption,
     defaultsTo: null,
     help: 'The name of the template to inject the code into.',
   );
   parser.addOption(
-    'input',
+    _kInputOption,
     defaultsTo: environment['INPUT'],
     help: 'The input file containing the snippet code to inject.',
   );
   parser.addOption(
-    'package',
+    _kPackageOption,
     defaultsTo: environment['PACKAGE_NAME'],
     help: 'The name of the package that this snippet belongs to.',
   );
   parser.addOption(
-    'library',
+    _kLibraryOption,
     defaultsTo: environment['LIBRARY_NAME'],
     help: 'The name of the library that this snippet belongs to.',
   );
   parser.addOption(
-    'element',
+    _kElementOption,
     defaultsTo: environment['ELEMENT_NAME'],
     help: 'The name of the element that this snippet belongs to.',
   );
 
   final ArgResults args = parser.parse(argList);
 
-  SnippetType snippetType;
-  for (SnippetType type in SnippetType.values) {
-    if (getEnumName(type) == args['type']) {
-      snippetType = type;
-    }
-  }
-  assert(snippetType != null, "Unable to find '${args['type']}' in SnippetType enum.");
+  final SnippetType snippetType = SnippetType.values
+      .singleWhere((SnippetType type) => getEnumName(type) == args[_kTypeOption]);
+  assert(snippetType != null, "Unable to find '${args[_kTypeOption]}' in SnippetType enum.");
```

Comment:
`singleWhere()` will throw if there's not a single element that matches, so this can't be `null` here.

otoh, asserts don't hurt 😄

Reply:
Actually, since `singleWhere` tests against all of the members, and because I'd rather have an assert with a message than an exception, I changed this to `firstWhere`, set to return null if nothing is found, and keep the assert.